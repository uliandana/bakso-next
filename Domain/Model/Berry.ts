export interface Berry {
  id: string,
  name: string,
  sprite: string,
  firmness: string,
}
