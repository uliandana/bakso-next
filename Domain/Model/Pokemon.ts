export interface Pokemon {
  id: string,
  name: string,
  nameSlug: string,
  sprite: string,
}
