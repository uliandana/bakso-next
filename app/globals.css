@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --foreground-rgb: 255, 255, 255;
  --background-start-rgb: 188,55,54;
  --background-end-rgb: 188,55,54;
  font-size: calc(768px * 16 / 375);
}

@media (prefers-color-scheme: dark) {
  :root {
    --foreground-rgb: 255, 255, 255;
    --background-start-rgb: 0, 0, 0;
    --background-end-rgb: 0, 0, 0;
  }
}

@layer utilities {
  .card-flip {
    position: absolute;
    backface-visibility: hidden;
    transform: rotateY(0);
    transition: transform 0.5s ease-in-out;
  }
  .card-flipped {
    position: absolute;
    backface-visibility: hidden;
    transform: rotateY(180deg);
    transition: transform 0.5s ease-in-out;
  }
  .silhouette {
    filter: brightness(0) drop-shadow(0.0625rem 0.125rem 0.25rem rgb(var(--background-start-rgb)));
  }
}

body {
  color: rgb(var(--foreground-rgb));
  background: linear-gradient(
      to bottom,
      transparent,
      rgb(var(--background-end-rgb))
    )
    rgb(var(--background-start-rgb));
}

* {
  animation: fade-in 0.5s ease-in-out;
}

@keyframes loading {
  to { transform: translateX(100%); }
}

@keyframes fade-in {
  from { opacity: 0; }
  to { opacity: 1; }
}

@media only screen and (max-width: 1440px) {
  :root {
    font-size: calc(100vw * 16 / 1440);
  }
}

@media only screen and (max-width: 768px) {
  :root {
    font-size: calc(100vw * 16 / 768);
  }
}
