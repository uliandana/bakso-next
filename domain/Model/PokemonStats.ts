export interface PokemonStats {
  name: string,
  value: number,
}
