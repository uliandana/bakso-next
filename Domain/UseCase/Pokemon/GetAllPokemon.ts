import { Pokemon } from '@/Domain/Model/Pokemon';
import { PokemonRepository } from '@/Domain/Repository/PokemonRepository';

export interface GetAllPokemonUseCase {
  invoke: (offset: number) => Promise<Pokemon[]>,
}

export class GetAllPokemon implements GetAllPokemonUseCase {
  private pokemonRepo: PokemonRepository;
  
  constructor(_pokemonRepo: PokemonRepository) {
    this.pokemonRepo = _pokemonRepo;
  }

  async invoke() {
    return this.pokemonRepo.getAllPokemon();
  }
}
