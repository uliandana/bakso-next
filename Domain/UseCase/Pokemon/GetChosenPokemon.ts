import { Pokemon } from '@/Domain/Model/Pokemon';
import { PokemonRepository } from '@/Domain/Repository/PokemonRepository';

export interface GetChosenPokemonUseCase {
  invoke: () => Promise<Pokemon['nameSlug']>,
}

export class GetChosenPokemon implements GetChosenPokemonUseCase {
  private pokemonRepo: PokemonRepository;
  
  constructor(_pokemonRepo: PokemonRepository) {
    this.pokemonRepo = _pokemonRepo;
  }

  async invoke() {
    return this.pokemonRepo.getChosenPokemon();
  }
}
