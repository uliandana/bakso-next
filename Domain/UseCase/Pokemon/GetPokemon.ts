import { Pokemon } from '@/Domain/Model/Pokemon';
import { PokemonRepository } from '@/Domain/Repository/PokemonRepository';

export interface GetPokemonUseCase {
  invoke: (offset: number) => Promise<Pokemon[]>,
}

export class GetPokemon implements GetPokemonUseCase {
  private pokemonRepo: PokemonRepository;
  
  constructor(_pokemonRepo: PokemonRepository) {
    this.pokemonRepo = _pokemonRepo;
  }

  async invoke(offset: number) {
    return this.pokemonRepo.getPokemon(offset);
  }
}
