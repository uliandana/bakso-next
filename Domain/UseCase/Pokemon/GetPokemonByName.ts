import { Pokemon } from '@/Domain/Model/Pokemon';
import { PokemonRepository } from '@/Domain/Repository/PokemonRepository';

export interface GetPokemonByNameUseCase {
  invoke: (name: string, statOnly?: Boolean) => Promise<Pokemon>,
}

export class GetPokemonByName implements GetPokemonByNameUseCase {
  private pokemonRepo: PokemonRepository;
  
  constructor(_pokemonRepo: PokemonRepository) {
    this.pokemonRepo = _pokemonRepo;
  }

  async invoke(name: string, statOnly?: Boolean) {
    return this.pokemonRepo.getPokemonByName(name, statOnly);
  }
}
