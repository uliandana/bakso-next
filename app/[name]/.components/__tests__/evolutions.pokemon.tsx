import React from 'react';
import { describe, expect, jest, test } from '@jest/globals';
import { Pokemon } from '@/domain/Model/Pokemon';
import ShallowRenderer from 'react-test-renderer/shallow';
import Evolutions from '../evolutions.pokemon';

const dummyPokemon: Pokemon = {
  id: '01',
  name: 'pikachu',
  nameSlug: 'pikachu',
  sprite: 'url',
  stats: [],
  types: ['bug'],
  baseWeight: 10,
  weight: 10,
  evolvesTo: [],
  bgColor: '#fff',
};

describe('./app/[name]/.components/evolutions.pokemon', () => {
  test('renders', () => {
    const shallow = new ShallowRenderer();
    const onSelect = jest.fn();
    const tree = shallow.render(<Evolutions evolutions={[dummyPokemon]} className="test-class" targetId="01" onSelect={onSelect} />);
    expect(tree).toMatchSnapshot();
  });

  test('render empty', () => {
    const onSelect = jest.fn();
    const props = {
      evolutions: [],
      className: '',
      targetId: '',
      onSelect,
    };
    const res = Evolutions(props);
    expect(res).toBe(null);
  });

  test('render multi evolutions and trigger onSelect', () => {
    const onSelect = jest.fn();
    const props = {
      evolutions: [dummyPokemon, dummyPokemon],
      className: '',
      targetId: '',
      onSelect,
    };
    const res = Evolutions(props);
    expect(res?.props.children.length).toBe(2);
    const e = {
      target: {
        value: '0',
      },
    };
    res?.props.children[0].props.children[0].props.onChange(e);
    expect(props.onSelect).toHaveBeenCalled();
  });
});
