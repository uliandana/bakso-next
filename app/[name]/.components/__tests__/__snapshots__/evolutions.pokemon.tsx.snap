// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./app/[name]/.components/evolutions.pokemon renders 1`] = `
<div
  className="test-class"
>
  <img
    alt="evolution"
    className="silhouette"
    src="url"
  />
</div>
`;
